### PR TITLE
Tweak and extend `switch` tests

### PR DESCRIPTION
**Pull request by ltratt**
The first two commits of this PR tweak the existing `switch` tests we have. Interestingly, one of those already gave me a clue for what the problem @ptersilie has seen elsewhere is:

```
//       switch i32 %{{cond}}, label %{{default-bb}} [
//         i32 299, label %guardfail...
//         i32 298, label %guardfail...
//       ]
```

In other words, we're translating `switch`es to `switch`es. It probably is possible to do this correctly, but it probably needs handling quite carefully.

From that guess, it was easy to spot the flaw in our tests: we have switches with many `case`es, but we only ever use one of them in tracing and subsequent execution. So https://github.com/ykjit/yk/commit/40c6a7be8fe2037c082710e62be67e85c3ecc522 creates a test which forces many (all!) branches to be traced and executed. Once tracing is turned on, it fails, and is thus currently `ignore`d.

### CODE REVIEW

**Comment by ltratt** [comment]
@ptersilie Oops, https://github.com/ykjit/yk/pull/855/commits/42711710d5b23dd04bc6ba36a6c61d28609ed21f will need squashing.

**Comment by ptersilie** [comment]
Please squash.

**Comment by ltratt** [comment]
I'm not sure what causes the segfault in https://github.com/ykjit/yk/pull/855/commits/e666a99ad36bf8df774dee946c5e109bd6e34a1f when tracing is turned on, but it might be a useful additional test (note: it doesn't crashes when tracing isn't turned on). OK to squash?

**Comment by ptersilie** [comment]
Please squash.

File: tests/c/switch_nested_guard.c [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  int k = 0;
+  NOOPT_VAL(i);
+  NOOPT_VAL(j);
+  while (i > 0) {
+    yk_mt_control_point(mt, &loc);
+    char c, d;
+    switch (j % 7) {
+    case 6:
+      switch (k % 5) {
+        case 5:
```
[the remark this answers is not in the document]
Comment (ltratt) [reply tests/c/switch_nested_guard.c]: You're right -- sloppy copy and paste error! I've just forced pushed but 5ce5ff7 removes this. OK to squash?

**Comment by ptersilie** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**